lib/pdk/util.rb

```diff
@@ -1,8 +1,38 @@
 require 'tmpdir'
 require 'tempfile'
+require 'fiddle/import' if Gem.win_platform?
 
 module PDK
   module Util
+    module WindowsFunctions
+      if Gem.win_platform?
+        extend Fiddle::Importer
+        dlload 'kernel32.dll'
+        extern('int GetLongPathNameA(char*, char*, int)', :stdcall)
+
+        # Call into kernel32.dll to resolve short path e.g. C:\Users\~ADMIN1\foo
+        # should be changed to long path i.e. C:\Users\Administrator\foo
+        def long_path(short_path)
+          # Fetch long path name into a buffer
+          bufferlen = 255
+          buffer    = nil
+          loop do
+            buffer = ' ' * bufferlen
+            len    = GetLongPathNameA(short_path.to_s, buffer, buffer.size)
```

Comment:
You need to use the W (`GetLongPathNameW`) version of the API plus be able to handle double byte characters.  Otherwise it'll break for non-English letters e.g. Jöhn

https://github.com/djberg96/win32-file/blob/ffi/lib/win32/file.rb#L189-L200

https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/file.rb#L267-L279